src/main/java/soot/dexpler/DexBody.java

```diff
@@ -204,6 +233,8 @@ protected DexBody(DexEntry<? extends DexFile> dexFile, Method method, RefType de
 
     instructions = new ArrayList<DexlibAbstractInstruction>();
     instructionAtAddress = new HashMap<Integer, DexlibAbstractInstruction>();
+    localDebugs = new HashMap<Integer, LinkedList<RegDbgEntry>>();
```

Comment:
That should be some sort of LinkedMultiMap (maybe create such a class if we don't have it already) for simplifying the code here and not having to deal with putting lists into maps

Reply:
Fixed with `ArrayListMultimap` from guava.